server/lib/queries.js

```diff
@@ -64,14 +68,18 @@ module.exports = function(sequelize) {
       minTotalDonationClause = ''
     }
 
+    if (tag) {
+      tagClause = 'g.tags && $tag AND';
+    }
+
     return sequelize.query(`
       WITH "totalDonations" AS (
         SELECT "GroupId", SUM(amount) as "totalDonations", MAX(currency) as currency, COUNT(DISTINCT "GroupId") as collectives FROM "Transactions" WHERE amount > 0 AND currency='USD' AND "PaymentMethodId" IS NOT NULL GROUP BY "GroupId"
       )
       SELECT g.id, g.name, g.slug, g.mission, g.logo, g."backgroundImage", t."totalDonations", t.currency, t.collectives
       FROM "Groups" g LEFT JOIN "totalDonations" t ON t."GroupId" = g.id
-      WHERE ${minTotalDonationClause} g.tags && $tag AND g."deletedAt" IS NULL ${excludeClause}
-      ORDER ${orderClause} DESC NULLS LAST LIMIT ${limit}
+      WHERE ${minTotalDonationClause} ${tagClause} g."deletedAt" IS NULL ${excludeClause}
+      ORDER ${orderClause} ${orderDirection} NULLS LAST LIMIT ${limit} OFFSET ${offset || 0}
```

Comment:
`mw.paginate` is hard coded to paginate using a table's `id` column, I don't think I can use it here.
https://github.com/OpenCollective/opencollective-api/blob/master/server/controllers/middlewares.js#L309

`mw.sorting` is usuable, however since we are using a raw query I don't believe I'll take full advantage of it, the middleware sets `req.paginate.dir`, which I would use the same way I use `req.query.sort` now, **plus** a function call to escape the input to prevent blind sql injection.

I believe these middlewares where intended to use alongside `_.merge()` and not-raw-sql: 
https://github.com/OpenCollective/opencollective-api/blob/master/server/controllers/groups.js#L140

Thanks for bringing these middlewares to light, I will make sure I take advantage of them from now on